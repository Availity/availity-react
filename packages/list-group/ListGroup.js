import React from 'react';
import PropTypes from 'prop-types';
import { ListGroup as RsListGroup } from 'reactstrap';

const ListGroup = React.forwardRef(
  ({ cards, selectable, className, ...props }, ref) => {
    const classname = [
      className,
      cards && 'list-group-cards',
      selectable && 'list-group-selectable',
    ]
      .filter(a => a)
      .join(' ');
    return <RsListGroup ref={ref} {...props} className={classname} />;
  }
);

ListGroup.propTypes = {
  cards: PropTypes.bool,
  selectable: PropTypes.bool,
  className: PropTypes.string,
};

export default ListGroup;
