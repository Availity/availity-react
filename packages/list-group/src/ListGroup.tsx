import React from 'react';
import { ListGroup as RsListGroup, ListGroupProps } from 'reactstrap';
import classNames from 'classnames';

type Props = {
  cards?: boolean;
  className?: string;
  selectable?: boolean;
} & ListGroupProps;

const ListGroup = React.forwardRef<RsListGroup, Props>(({ cards, selectable, className, ...props }, ref) => (
  <RsListGroup
    ref={ref}
    {...props}
    className={classNames(className, { 'list-group-cards': cards, 'list-group-selectable': selectable })}
  />
));

export default ListGroup;
