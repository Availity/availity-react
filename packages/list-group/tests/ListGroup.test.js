import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ListGroup from '..';

afterEach(cleanup);

describe('ListGroup', () => {
  test('should render', () => {
    const { container } = render(<ListGroup />);

    expect(container).toMatchSnapshot();
  });

  test('should render cards', () => {
    const { container } = render(<ListGroup cards />);

    expect(container).toMatchSnapshot();
  });

  test('should render selectable', () => {
    const { container } = render(<ListGroup selectable />);

    expect(container).toMatchSnapshot();
  });

  test('should render selectable cards', () => {
    const { container } = render(<ListGroup selectable cards />);

    expect(container).toMatchSnapshot();
  });
});
