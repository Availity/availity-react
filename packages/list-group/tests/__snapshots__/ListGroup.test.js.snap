// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup should render 1`] = `
<div>
  <ul
    class="list-group"
  />
</div>
`;

exports[`ListGroup should render cards 1`] = `
<div>
  <ul
    class="list-group-cards list-group"
  />
</div>
`;

exports[`ListGroup should render selectable 1`] = `
<div>
  <ul
    class="list-group-selectable list-group"
  />
</div>
`;

exports[`ListGroup should render selectable cards 1`] = `
<div>
  <ul
    class="list-group-cards list-group-selectable list-group"
  />
</div>
`;
